tools/tensorflow/tensor_run.py

```diff
@@ -0,0 +1,43 @@
+import json
+import sys
+import time
+from importlib import import_module
+
+import tensorflow as tf
+
+
+def resolve(name):
+    functions = import_module("tensor_functions")
+    return getattr(functions, name)
+
+
+def tensor(x):
+    return tf.Variable(x, dtype=tf.float64)
+
+
+def run(params):
+    func = resolve(params["name"])
+    vals = [tensor(arg["value"]) for arg in params["arguments"]]
+    
+    
+    with tf.GradientTape() as tape:
+        tape.watch(vals)
+        ret = func(*vals)
+        
+    start = time.perf_counter_ns()
+    gradients = tape.gradient(ret,vals)
```

Comment:
As mentioned in `tensor_functions.py`, this code shouldn't actually take the gradient; it should just call the function, and the implementation of the function it calls should be in charge of taking gradients.